c10/util/Float8_e4m3fn.h

```diff
@@ -14,14 +14,11 @@
 /// Implementation based on the paper https://arxiv.org/pdf/2209.05433.pdf
 /// and inspired by Half implementation from pytorch/c10/util/Half.h
 
+#include <c10/macros/Export.h>
 #include <c10/macros/Macros.h>
-#include <c10/util/C++17.h>
-#include <c10/util/TypeSafeSignMath.h>
 #include <c10/util/floating_point_utils.h>
-#include <type_traits>
 
 #if defined(__cplusplus) && (__cplusplus >= 201103L)
-#include <cmath>
```

Comment:
Are we sure these headers are not needed? We had a few PRs open recently like https://github.com/pytorch/pytorch/pull/116586 so I want to make sure we aren't going overboard for rarer compilers.